launcher/emitter.go

```diff
@@ -95,3 +115,38 @@ func GetCompletedTaskOutput(taskId int) ([]string, error) {
 	}
 	return lines, nil
 }
+
+func PullS3TaskOutput(taskId int) error {
+	tm := GetTaskManager()
+	task := tm.Get(taskId)
+	if task == nil {
```

Comment:
There is a "Christmas tree" problem with this code (if inside if....). Please slack me to elaborate.